### PR TITLE
Functor thermal solid properties

### PR DESCRIPTION
**Pull request by GiudGiud**
For @tanoret

### CODE REVIEW

File: modules/solid_properties/src/materials/ThermalSolidPropertiesFunctorMaterial.C [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  return params;
+}
+
+ThermalSolidPropertiesFunctorMaterial::ThermalSolidPropertiesFunctorMaterial(
+    const InputParameters & parameters)
+  : FunctorMaterial(parameters),
+    _temperature(getFunctor<ADReal>("temperature")),
+    _sp(getUserObject<ThermalSolidProperties>("sp"))
+{
+  addFunctorProperty<ADReal>(getParam<std::string>(SolidPropertiesNames::specific_heat),
```
Comment (GiudGiud) [inline modules/solid_properties/src/materials/ThermalSolidPropertiesFunctorMaterial.C:41]: given the new conversion capability I dont think there s a need to template those on AD anymore

File: modules/solid_properties/test/tests/materials/thermal_solid_functor_properties/tests [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[Tests]
+  [test]
+    type = CSVDiff
+    input = 'test.i'
+    csvdiff = 'test_out.csv'
+    requirement = "The system shall provide the ability to use functor for solid material "
```
Comment (joshuahansel) [inline modules/solid_properties/test/tests/materials/thermal_solid_functor_properties/tests:6]:
```suggestion
    requirement = "The system shall provide the ability to use functors for solid material "
```

**Review by joshuahansel — CHANGES_REQUESTED** [review]
Just the typo.

**Comment by GiudGiud** [comment]
Squashed in

**Review by joshuahansel — APPROVED** [review]

**Pull request merged**